### PR TITLE
Reenable iam tests with recordings

### PR DESCRIPTION
**Pull request by VenelinMartinov**
Recorded snapshots for all the tests and re-enabled them.

Should address https://github.com/pulumi/pulumi-gcp/issues/1500

The `Cluster` test takes 12 minutes to pulumi up. Is this worth doing in CI?

### CODE REVIEW

**Comment by t0yv0** [comment]
> The Cluster test takes 12 minutes to pulumi up. Is this worth doing in CI?

What does it mean "worth doing in CI", can you elaborate? AFAIK testing cluster does not require `pulumi up`. Recording does but it is less frequent.

**Comment by VenelinMartinov** [comment]
Looks like the recordings have the google project:
```
Expected no changes for {urn:pulumi:p-it-runner-iam-member-83a2add9::iam-member::gcp:projects/iAMMember:IAMMember::projectIAMMember true [project] [project] false map[project:pulumi-development] map[project:pulumi-ci-gcp-provider]
```
which causes the tests to fail, since I recorded using the dev project but CI has a separate one.

**Comment by t0yv0** [comment]
> which causes the tests to fail, since I recorded using the dev project but CI has a separate one.

The easiest fix is to record using the same project as CI.  There might be some workarounds.

**Comment by t0yv0** [comment]
Like you can specialize the "expected no changes" check to permit this change but look out for others. Or try to stick this property in ingoreChanges.

File: provider/resources.go
Comment (guineveresaenger) [inline provider/resources.go:1]: These are really noisy; could we not commit them at least until we enforce Go formatting?

**Review by guineveresaenger — APPROVED** [review]
Thank you for getting these in!

**Comment by t0yv0** [comment]
💥

**Pull request merged**